app/api/services/users/patch_user.py

```diff
@@ -23,69 +25,50 @@ class PatchUserParams(TypedDict, total=False):
 # https://github.com/navapbc/template-application-flask/issues/49#issue-1505008251
 # TODO: Use classes / objects as inputs to service methods
 # https://github.com/navapbc/template-application-flask/issues/52
-def patch_user(user_id: str, patch_user_params: PatchUserParams, api_context: ApiContext) -> User:
-    # TODO: move this to service and/or persistence layer
-    user = api_context.db_session.query(User).options(orm.selectinload(User.roles)).get(user_id)
-
-    if user is None:
-        # TODO move HTTP related logic out of service layer to controller layer and just return None from here
-        # https://github.com/navapbc/template-application-flask/pull/51#discussion_r1053754975
-        raise apiflask.HTTPError(404, message=f"Could not find user with ID {user_id}")
-
-    for key, value in patch_user_params.items():
-        if key == "roles":
-            _handle_role_patch(user, patch_user_params["roles"], api_context)
-            continue
+def patch_user(
+    db_session: Session,
+    user_id: str,
+    patch_user_params: PatchUserParams,
+) -> User:
+
+    with db_session.begin():
+        # TODO: move this to service and/or persistence layer
+        user = db_session.query(User).options(orm.selectinload(User.roles)).get(user_id)
+
+        if user is None:
+            # TODO move HTTP related logic out of service layer to controller layer and just return None from here
+            # https://github.com/navapbc/template-application-flask/pull/51#discussion_r1053754975
+            raise apiflask.HTTPError(404, message=f"Could not find user with ID {user_id}")
+
+        for key, value in patch_user_params.items():
+            if key == "roles":
+                _handle_role_patch(db_session, user, patch_user_params["roles"])
+                continue
+
+            setattr(user, key, value)
+    return user
```

Comment:
Might be better to handle this with the persistence layer work that'll come later, but I'm curious if the services should be structured like this, since every single one will follow this pattern:

```py
with db_session.begin():
      # the service logic

return response
```

Feels like the session should begin a layer up, which also helps this be slightly less nested. When the persistence layer is added, I'd assume we don't want any DB logic here

Fine to wait until that is tackled to tinker with this more.

Reply:
Yeah kind of agree, we can tinker with that in a subsequent PR